user_modules/shared_testcases/commonFunctions.lua

```diff
@@ -51,7 +52,78 @@ function commonFunctions:isMediaApp()
   end
 
   return isMedia
+end
+
+--check that SDL ports are open then raise else RUN after timeout configured by step variable
+function commonFunctions:waitForSDLStart(test)
+  local step = 100
+  local hmiPort = config.hmiPort
+  local event = events.Event()
+  event.matches = function(self, e) return self == e end
+  local function raise_event()
+    assert(hmiPort ~= nil or hmiPort ~= "")
+    local output = os.execute ("netstat -vatn  | grep " .. hmiPort .. " | grep LISTEN")
+    if (output) then
+      RAISE_EVENT(event, event)
+    else 
+      RUN_AFTER(raise_event, step)
+    end
+  end
+  RUN_AFTER(raise_event, step)
+  local ret = expectations.Expectation("Wait for SDL start", test.mobileConnection)
+  ret.event = event
+  event_dispatcher:AddEvent(test.mobileConnection, ret.event, ret)
+  test:AddExpectation(ret)
+  return ret
+end
+
+function commonFunctions:createMultipleExpectationsWaiter(test, name)
+  local expectations = require('expectations')
```

Comment:
@LuxoftAKutsan I agree with you in easy moving this code block part. But my opinion is next: if this approach will take a mass effect resolving dependencies task become harder.